### DISCUSSION TITLE
Disconnect tm1 session when complete

### DISCUSSION THREAD

**Discussion by swartz2011** [cubewise-code/tm1py#529] [Q&A] [answered]
Is there a feature in TM1py where I can disconnect or end my session for a specific task?

for example
Login: my_single_connection = TM1Service(**config[tm1_source_server], timeout=1500)

if I am done before 1500 seconds, my system user stays logged in and idle until timeout is reached

Can I code something to "logout" upon finishing before 1500 seconds is reached?

**Comment by rkvinoth** [comment, marked as answer]
```python
my_single_connection.logout()
```

Should log you out.

Alternatively you can create a context manager.

```python
with TM1Service() as my_single_connection:
    #Your code goes here
```
